### PR TITLE
Convert vsintegration projects to .NET Core SDK

### PR DESCRIPTION
**Pull request by brettfo**
Eventually we'll properly use `Directory.Build.props` and `Directory.Build.targets`, but for the time being I don't want the non-SDK projects contaminated, so I opted to name everything `FSharp.Directory.Build.props/.targets` and opt-in each project by adding it's own `Directory.Build.props/.targets` next to the converted project files that redirects to the `FSharp.` ones.

To properly handle restoring `<PackageReference />` items, `NuGet.config` had to be moved to the root of the repo so that it's always up-path from every project file.

Also, to enable overriding the F# SDK targets, the `-proto` suffix had to be removed.  The proto compiler is still built into it's own directory, though, so there won't be any conflicts.

All projects under the `vsintegration/` directory have been converted to the new SDK **except** the property pages because that was giving me no end of grief and ultimately the projects are being converted to enable daily testing of the new project system work and the property pages project is VB anyways.

### CODE REVIEW

**Comment by brettfo** [comment]
@dotnet-bot test Ubuntu14.04 Release_default Build please

**Comment by brettfo** [comment]
@dotnet-bot test Ubuntu14.04 Release_fcs Build please

**Comment by dsyme** [comment]
@dotnet-bot test Ubuntu16.04 Release_default Build please

**Comment by dsyme** [comment]
image

**Comment by brettfo** [comment]
@smoothdeveloper What version of Visual Studio and MSBuild do you have installed?  As for the 14.0 and 15.0 versions of the Microsoft.VisualStudio.Shell package, both are needed, but duplicate type information should be handled by the `PrivateAssets="all"` on the first package reference and the aliasing of the 14.0 assembly [here](https://github.com/Microsoft/visualfsharp/blob/ff078e94deff66e548efb668465fcdd601cc158d/vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectSystem.Base.csproj#L81).

Also, have you cleaned the repo?  There might be some stuff left over in the `obj/` directory.

**Comment by smoothdeveloper** [comment]
@brettfo thanks for the explanations.

I'm using 

```
C:\dev\projects\github.com\microsoft\visualfsharp2>msbuild /version
Microsoft (R) Build Engine version 15.6.82.30579 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

15.6.82.30579
```

I tried from a clean clone and still having the errors: [buildlog.txt](https://github.com/Microsoft/visualfsharp/files/1857473/buildlog.txt)

**Comment by brettfo** [comment]
tl;dr - The evaluation order of targets appears to have changed.  I'm looking into it now.

@smoothdeveloper Now I'm seeing the same thing on my box.  Doing a quick look it appears that the `ResolveReferences` target is being invoked by `CoreBuild`, but that's happening after the `Compile` target which is what actually invokes `csc`.  In short, `Compile` is happening _before_ `ResolveReferences` which seems wrong to me.
